modules/mqtt/destination/mqtt-worker.c

```diff
@@ -202,6 +220,9 @@ _connect(LogThreadedDestWorker *s)
       conn_opts.httpsProxy = owner->http_proxy;
     }
 
+  MQTTClient_SSLOptions ssl_opts = _create_ssl_options(self);
+  conn_opts.ssl = &ssl_opts;
```

Comment:
Can this conflict with username/password? Also the same question is relevant here about setting these unconditionally / pre-checking the necessary values init time.

Reply:
User/pass authentication works well with TLS. Actually, it is the preferred way, otherwise, the password would be sent as plaintext.

(Using mutual authentication in TLS makes this redundant, of course.)